pkg/testingsdk/k8sclient.go

```diff
@@ -0,0 +1,652 @@
+package testingsdk
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	stov1 "k8s.io/api/storage/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"strings"
+	"time"
+)
+
+var (
+	ResourceExisted            = errors.New("[REX] resource is already existed")
+	ResourceNotFound           = errors.New("[RNF] resource is not found")
+	ResourceNameNull           = errors.New("[RNN] resource name is null")
+	ControlPlaneLabel          = "node-role.kubernetes.io/control-plane"
+	defaultRetryInterval       = 10 * time.Second
+	defaultRetryTimeout        = 160 * time.Second
+	defaultLongRetryInterval   = 20 * time.Second
+	defaultLongRetryTimeout    = 300 * time.Second
+	defaultNodeInterval        = 2 * time.Second
+	defaultNodeReadyTimeout    = 20 * time.Minute
+	defaultNodeNotReadyTimeout = 8 * time.Minute
+)
+
+func isPvcReady(ctx context.Context, k8sClient *kubernetes.Clientset, nameSpace string, pvcName string) error {
+	err := wait.PollImmediate(defaultRetryInterval, defaultRetryTimeout, func() (bool, error) {
+		ready := false
+		pvc, err := getPVC(ctx, k8sClient, nameSpace, pvcName)
+		if err != nil {
+			if IsRetryableError(err) {
+				return false, nil
+			}
+			return false, fmt.Errorf("unexpected error occurred while getting pvc [%s]", pvcName)
+		}
+		if err != nil {
+			return false, nil
+		}
+		if pvc != nil && pvc.Status.Phase == apiv1.ClaimBound {
+			ready = true
+		}
+		if !ready {
+			fmt.Printf("pvc %s is not bound\n", pvc.Name)
+			return false, nil
+		}
+		return true, nil
+	})
+	return err
+}
+
+func isDeploymentReady(ctx context.Context, k8sClient *kubernetes.Clientset, nameSpace string, deployName string) error {
```

Comment:
same: this is also a `waitFor....`

I believe the other ones with the PollImmediate are also like that